client/src/js/directives/bhFindSupplier.js

```diff
@@ -0,0 +1,201 @@
+angular.module('bhima.directives')
+.component('bhFindSupplier', {
+  controller: FindSupplierComponent,
+  templateUrl : 'partials/templates/bhFindSupplier.tmpl.html',
+  bindings: {
+    onSearchComplete: '&',  // bind callback
+    type:             '@',  // bind string
+    required:         '<',  // bind the required
+  }
+});
+
+FindSupplierComponent.$inject = ['SupplierService', 'appcache'];
+
+/**
+ * The Find Supplier Component
+ *
+ * This component allows a user to serach for a supplier by either the
+ * supplier identifier (Project Abbreviation + Reference) or by typeahead on supplier
+ * name.
+ *
+ * The typeahead loads data as your type into the input box, pinging th URL
+ * /supplier/search/?name={string} The HTTP endpoints sends back 20 results
+ * which are presented to the user.
+ *
+ * SUPPORTED ATTRIBUTES:
+ *   - type : which take one of these values (inline or panel) (default: inline)
+ *   - on-search-complete : the callback function which get the returned supplier
+ */
+function FindSupplierComponent(SupplierService, AppCache) {
+  var vm = this;
+
+  /** cache to remember which the search type of the component */
+  var cache = new AppCache('FindSupplierComponent');
+
+  /** @const the max number of records to fetch from the server */
+  var LIMIT = 20;
+
+  /** supported searches: by name or by id */
+  vm.options = {
+    findByName : {
+      'label' : 'FIND.SUPPLIER_NAME',
+      'placeholder' : 'FIND.SEARCH_NAME'
+    }
+  };
+
+  vm.timestamp      = new Date();
+  vm.showSearchView = true;
+  vm.loadStatus     = null;
+  vm.validInput     = false;
+
+  /** Expose functions and variables to the template view */
+  vm.searchByName       = searchByName;
+  vm.selectSupplier      = selectSupplier;
+
+  vm.validateNameSearch = validateNameSearch;
+  vm.findBy             = findBy;
+  vm.reload             = reload;
+  vm.readInput          = readInput;
+
+  /** fetch the initial setting for the component from appcache */
+  cache.fetch('optionKey')
+  .then(loadDefaultOption);
+
+
+  /**
+  * @method searchByName
+  *
+  * @param {string} text Supplier name 
+  *
+  * @description This function make a call to BHIMA API for getting suppliers
+  * according the name of supplier.
+  *
+  * @return {Array} An array of suppliers
+  */
+  function searchByName(text) {
+    vm.loadStatus = 'loading';
+
+    // format query string parameters
+    var options = {
+      name : text.toLowerCase(),
+      limit : LIMIT
+    };
+
+    return SupplierService.filter(options)
+    .then(function (suppliers) {
+      // loop through each
```

Comment:
If there is only one property for the name of the supplier, you do not need to cache the label.